### PR TITLE
Fix JIT/SPMI CodeQL issues

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Fixes the following CodeQL issues:
* https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1664873
* https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1664968
* https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1664926
* https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1664940

@dotnet/jit-contrib PTAL, thanks!

### CODE REVIEW

File: src/coreclr/tools/superpmi/superpmi/icorjitinfo.cpp [hunk 1]
```diff
@@ -1775,7 +1775,7 @@ int MyICJI::doAssert(const char* szFile, int iLine, const char* szExpr)
     sprintf_s(buff, sizeof(buff), "%s (%d) - %s", szFile, iLine, szExpr);
 
     LogIssue(ISSUE_ASSERT, "#%d %s", jitInstance->mc->index, buff);
-    jitInstance->mc->cr->recMessageLog(buff);
+    jitInstance->mc->cr->recMessageLog(buff, "%s");
```
Comment (jakobbotsch) [inline src/coreclr/tools/superpmi/superpmi/icorjitinfo.cpp:1778]:
```suggestion
    jitInstance->mc->cr->recMessageLog("%s", buff);
```

File: src/coreclr/jit/lowerxarch.cpp [hunk 2]
```diff
@@ -4845,7 +4845,7 @@ GenTree* Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
         case TYP_INT:
         case TYP_UINT:
         {
-            unsigned       N            = 0;
```
Comment (jakobbotsch) [inline src/coreclr/jit/lowerxarch.cpp:4848]: Can it be declared inside the loops?

Comment (amanasifkhalid) [reply src/coreclr/jit/lowerxarch.cpp:4848]: Sure.

**Review by jakobbotsch — APPROVED** [review]

**Comment by amanasifkhalid** [comment]
/ba-g widespread timeouts not captured by Build Analysis

**Pull request merged**